servers/physics_3d/godot_area_3d.cpp

```diff
@@ -230,77 +230,75 @@ void GodotArea3D::set_monitorable(bool p_monitorable) {
 }
 
 void GodotArea3D::call_queries() {
-	if (monitor_callback_id.is_valid() && !monitored_bodies.is_empty()) {
-		Variant res[5];
-		Variant *resptr[5];
-		for (int i = 0; i < 5; i++) {
-			resptr[i] = &res[i];
-		}
+	if (!monitor_callback.is_null() && !monitored_bodies.is_empty()) {
+		if (monitor_callback.is_valid()) {
+			Variant res[5];
+			Variant *resptr[5];
+			for (int i = 0; i < 5; i++) {
+				resptr[i] = &res[i];
+			}
 
-		Object *obj = ObjectDB::get_instance(monitor_callback_id);
-		if (!obj) {
-			monitored_bodies.clear();
-			monitor_callback_id = ObjectID();
-			return;
-		}
+			for (Map<BodyKey, BodyState>::Element *E = monitored_bodies.front(); E;) {
+				if (E->get().state == 0) { // Nothing happened
+					Map<BodyKey, BodyState>::Element *next = E->next();
+					monitored_bodies.erase(E);
```

Comment:
Not really related to this PR, but should be element `E` from map `monitoring_bodies` removed when iterating over this map?

Reply:
I'm not sure. Is true that it doesn't look very safe. Would be good to have the opinion of somebody with more experience. If changes need to be done to make it safe, probably I can include those in this same PR

Reply:
Iteration is handled in the next line with `E = next` so it looks like it's safe.